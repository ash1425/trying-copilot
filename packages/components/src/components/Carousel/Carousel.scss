.carousel {
  position: relative;
  top: 10px;
}

.carousel-controls {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  left: 3%;
  position: absolute;
  right: 3%;
  top: 50%;
}

.circle {
  align-content: center;
  background: rgba(255, 255, 255, 0.8);
  border-radius: 70px;
  border-radius: 70px;
  border-radius: 70px;
  cursor: pointer;
  height: 70px;
  position: relative;
  text-align: center;
  width: 70px;
}

.circle:hover {
  opacity: 1;
}

.text {
  align-content: center;
  height: 50px;
  left: 15%;
  position: absolute;
  text-align: center;
  top: 35%;
  width: 50px;
}
