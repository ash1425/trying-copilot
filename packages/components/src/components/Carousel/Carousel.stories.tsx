import React from "react";

import { Meta, Story } from "@storybook/react";

import Carousel, { ICarouselProps } from "./Carousel";
import imageOne from "./images/alessandro-carrarini-YKa1kxa9YQo-unsplash.jpg";
import imageTwo from "./images/enrica-tancioni-kvpRkt9E1D4-unsplash.jpg";

export default {
  component: Carousel,
  title: "components/Carousel",
  argTypes: {
    className: {
      options: "carousel",
      control: { type: "text" },
    },
    data: {
      options: [
        {
          image: imageOne,
        },
        {
          image: imageTwo,
        },
      ],
      control: "array",
    },
  },
} as Meta;

const Template: Story<ICarouselProps> = (args) => {
  const { className, data } = args;
  return <Carousel className={className} data={data} />;
};
export const Default = Template.bind({});

Default.args = {
  className: "carousel",
  data: [
    {
      image: imageOne,
    },
    {
      image: imageTwo,
    },
  ],
};
