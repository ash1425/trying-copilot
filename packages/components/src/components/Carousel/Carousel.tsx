import React, { useState } from "react";
import "./Carousel.scss";

export interface ICarouselProps {
  className: string;
  data: {
    image: string;
  }[];
}

const Carousel = ({ className, data }: ICarouselProps) => {
  const [current, setCurrent] = useState(0);
  const { length } = data;
  const nextSlide = () => {
    setCurrent(current === length - 1 ? 0 : current + 1);
  };
  const prevSlide = () => {
    setCurrent(current === 0 ? length - 1 : current - 1);
  };
  return (
    <div>
      {data.map((image, index) => (
        <div key={`${current + index}`}>
          {index === current && (
            <div>
              <div className={`${className}`}>
                <div className={`${className}-image`}>
                  <img
                    src={image.image}
                    style={{ width: "100%", height: 400 }}
                    alt=""
                    id={`image-${index + 1}`}
                  />
                </div>

                <div className={`${className}-controls`}>
                  <div className="circle" onClick={prevSlide}>
                    <div className="text">prev</div>
                  </div>
                  <div className="circle" onClick={nextSlide}>
                    <div className="text">next</div>
                  </div>
                </div>
              </div>
            </div>
          )}
        </div>
      ))}
    </div>
  );
};
export default Carousel;
