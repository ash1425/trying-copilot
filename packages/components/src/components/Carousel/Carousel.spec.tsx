import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Carousel from "./Carousel";

import imageOne from "./images/alessandro-carrarini-YKa1kxa9YQo-unsplash.jpg";
import imageTwo from "./images/enrica-tancioni-kvpRkt9E1D4-unsplash.jpg";

describe("Carousel", () => {
  const ImageData = [
    {
      image: imageOne,
    },
    {
      image: imageTwo,
    },
  ];

  test("renders Carousel component", () => {
    render(<Carousel className="carousel" data={ImageData} />);
  });

  test("should have carousal content", () => {
    render(<Carousel className="carousel" data={ImageData} />);
    expect(screen.getByText("prev")).toHaveTextContent("prev");
    expect(screen.getByText("next")).toHaveTextContent("next");
  });

  test("when click on next button, get next slide", () => {
    render(<Carousel className="carousel" data={ImageData} />);
    userEvent.click(screen.getByText("next"));
    const image = screen.getByRole("img");
    expect(image).toHaveAttribute("id", "image-2");
    userEvent.click(screen.getByText("next"));
    const imagenext = screen.getByRole("img");
    expect(imagenext).toHaveAttribute("id", "image-1");
  });
  test("when click on prev button, get previous slide", () => {
    const { asFragment } = render(
      <Carousel className="carousel" data={ImageData} />
    );
    userEvent.click(screen.getByText("prev"));
    const image = screen.getByRole("img");
    expect(image).toHaveAttribute("id", "image-2");
    userEvent.click(screen.getByText("prev"));
    const imageprev = screen.getByRole("img");
    expect(imageprev).toHaveAttribute("id", "image-1");
    expect(asFragment()).toMatchSnapshot();
  });
});
