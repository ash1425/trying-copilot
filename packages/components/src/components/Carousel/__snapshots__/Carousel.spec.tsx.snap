// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel when click on prev button, get previous slide 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <div
          class="carousel"
        >
          <div
            class="carousel-image"
          >
            <img
              alt=""
              id="image-1"
              src="test-file-stub"
              style="width: 100%; height: 400px;"
            />
          </div>
          <div
            class="carousel-controls"
          >
            <div
              class="circle"
            >
              <div
                class="text"
              >
                prev
              </div>
            </div>
            <div
              class="circle"
            >
              <div
                class="text"
              >
                next
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
